### PR TITLE
Issue #794 by Inlead: Fix sorting and display of long-term events.

### PR DESCRIPTION
**Pull request by guddo**
http://platform.dandigbib.org/issues/794

### CODE REVIEW

File: modules/ding_frontend/ding_frontend.module [hunk 1]
```diff
@@ -67,6 +67,13 @@ function ding_frontend_field_formatter_view($entity_type, $entity, $field, $inst
 
   foreach ($items as $delta => $item) {
     if ($display['type'] == 'calendar_header') {
+      // Display today date if event continues (current date is between start and end date of an event).
+      if ($entity_type == 'node' && $entity->type == 'ding_event') {
+        $now = format_date(REQUEST_TIME, 'custom', 'Y-m-d H:i:s');
+        if (strtotime($item['value']) <= REQUEST_TIME && strtotime($item['value2']) >= REQUEST_TIME) {
+          $item['value'] = $now;
```
Comment (kasperg) [inline modules/ding_frontend/ding_frontend.module]: We might as well remove L72 and change L74 to:

```php
$item['value'] = format_date(REQUEST_TIME, 'custom', 'Y-m-d H:i:s');
```

Also: How do we know that `Y-m-d H:i:s` is the correct format for the item value?

File: modules/ding_event/ding_event.views.inc [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<?php
```
Comment (kasperg) [inline modules/ding_event/ding_event.views.inc]: Where/how is this file loaded?

File: modules/ding_event/views/ding_event_handler_sort_ding_event_date.inc [hunk 3]
```diff
@@ -0,0 +1,41 @@
+<?php
+
+/**
+ * @file
+ * Definition of ding_event_handler_sort_ding_event_date.
```
Comment (kasperg) [inline modules/ding_event/views/ding_event_handler_sort_ding_event_date.inc]: Please add more details about what this sorting is about.

**Comment by kasperg** [comment]
Reviewed. A few minor comments. Also remember to fix [the code style issues](https://scrutinizer-ci.com/g/ding2/ding2/inspections/6438bedc-7e0c-41e3-9cd0-5264926d3208/issues/)

File: modules/ding_frontend/ding_frontend.module (hunk 1 above)
Comment (koykov) [inline modules/ding_frontend/ding_frontend.module]: Y-m-d H:i:s is a default format for dates in development. $item['value'] contains the date in the same format.

File: modules/ding_event/ding_event.views.inc (hunk 2 above)
Comment (koykov) [inline modules/ding_event/ding_event.views.inc]: This file included automatically if hook_views_api() is implemented.

**Comment by koykov** [comment]
kasperg, from code style issues were ignored the following comments:
> Class name must begin with a capital letter
> Class name must use UpperCamel naming without underscores

views module has own class naming rules (see views_module_dir/handlers contents).

File: modules/ding_event/ding_event.views.inc (hunk 2 above)
Comment (kasperg) [inline modules/ding_event/ding_event.views.inc]: :+1:

**Comment by kasperg** [comment]
@koykov: Thanks and bunch for the update and for the elaboration about disregarding code style issues. It helps a lot and I agree.

Reviewed and approved.

**Comment by kasperg** [comment]
Merged. Thanks!

**Pull request merged**